File: fpsCounter.py
import time

TARGET_FPS = 60

def main():
    last_time = time.time()
    ticks = 0
    frames = 0

    while True:
        now = time.time()
        elapsed = now - last_time

        if elapsed >= 1.0:
            print(f"Ticks: {ticks}, Frames: {frames}")
            ticks = 0
            frames = 0
            last_time = now

        # Simulate game logic here (tick function)

        ticks += 1

        # Simulate frame rendering here (render function)

        frames += 1

        # Adjust frame rate here if needed to maintain TARGET_FPS

if __name__ == "__main__":
    main()

File: FpsCounter.c
#include <stdio.h>
#include <time.h>

#define TARGET_FPS 60

int main() {
    clock_t lastTime = clock();
    int ticks = 0;
    int frames = 0;

    while (1) {
        clock_t now = clock();
        double elapsed = (double)(now - lastTime) / CLOCKS_PER_SEC;

        if (elapsed >= 1.0) {
            printf("Ticks: %d, Frames: %d\n", ticks, frames);
            ticks = 0;
            frames = 0;
            lastTime = now;
        }

        // Simulate game logic here (tick function)

        ticks++;

        // Simulate frame rendering here (render function)

        frames++;

        // Adjust frame rate here if needed to maintain TARGET_FPS
    }

    return 0;
}
